server/src/test/java/org/elasticsearch/snapshots/SnapshotShutdownProgressTrackerTests.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.snapshots;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.coordination.Coordinator;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.index.snapshots.IndexShardSnapshotStatus;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.MockLog;
+import org.elasticsearch.threadpool.TestThreadPool;
+import org.junit.After;
+import org.junit.Before;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SnapshotShutdownProgressTrackerTests extends ESTestCase {
+    private static final Logger logger = LogManager.getLogger(SnapshotShutdownProgressTrackerTests.class);
+
+    private final ClusterService mockClusterService = mock(ClusterService.class);
+    private final ClusterState mockClusterState = mock(ClusterState.class);
+    private final DiscoveryNodes mockDiscoveryNodes = mock(DiscoveryNodes.class);
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        when(mockClusterService.state()).thenReturn(mockClusterState);
+        when(mockClusterState.nodes()).thenReturn(mockDiscoveryNodes);
+        when(mockDiscoveryNodes.getLocalNodeId()).thenReturn("localNodeIdForTest");
+    }
+
+    final Settings settings = Settings.builder()
+        .put(
+            SnapshotShutdownProgressTracker.SNAPSHOT_PROGRESS_DURING_SHUTDOWN_INTERVAL_TIME_SETTING.getKey(),
+            TimeValue.timeValueMillis(200)
+        )
+        .build();
+    TestThreadPool testThreadPool;
```

Comment:
Since this is the only thing scheduling any future tasks in the context of this test I think you should be able to just `advanceTime()` once and then `runAllRunnableTasks()` to trigger the next check. I would expect `runTasksUpToTimeInOrder()` to work too, not sure what problems you faced there. Note it takes the target "absolute" time, rather than the number of simulated milliseconds to allow to elapse?